pandas/indexes/base.py

```diff
@@ -1745,29 +1745,13 @@ def argsort(self, *args, **kwargs):
         return result.argsort(*args, **kwargs)
 
     def __add__(self, other):
-        if is_list_like(other):
-            warnings.warn("using '+' to provide set union with Indexes is "
-                          "deprecated, use '|' or .union()", FutureWarning,
-                          stacklevel=2)
-        if isinstance(other, Index):
```

Comment:
Not sure I understand this comment. We had indeed tests that tested the union behaviour of `+` for Index, and I changed this test to check for concatenation. If there are other tests that would hit this path, they should now fail given that I removed it.